android/Drawer/app/src/main/java/com/example/drawer/CanvasGrid.java

```diff
@@ -131,8 +184,28 @@ public boolean onTouchEvent(MotionEvent event) {
             int row = (int)(event.getY() / cellLength);
 
             try {
-                cellChecked[column][row] = true;
-            } catch (Exception e) {
+
+                if (cellChecked[column][row] != true) {
+                    cellChecked[column][row] = true;
+                    pureCellChecked[column][row] = true;
+
+                    if( firstTouch == false && (Math.abs(row - lastx) > 1 || Math.abs(column - lasty) > 1 ) ) {
```

Comment:
It is unclear in this statement which boolean values are supposed to be in what order, the `((statement && statement) || statement)`, or `(statement && (statement || statement))`. for readability, a set of parentheses could be introduced.